### ISSUE TITLE
BeginDisable doesn't disable shortcuts

### ISSUE THREAD

**Issue by rodrigorc** [ocornut/imgui#7726]
### Version/Branch of Dear ImGui:

Version: 1.90.8, Branch: master

### Back-ends:

all

### Compiler, OS:

all

### Details:

**Should `BeginDisable()` disable keyboard shortcuts?**

I had this happy code:
```
bool ok1 = Button("OK");
bool ok2 = Shortcut(ImGuiKey_Enter, ImGuiInputFlags_None);
if (ok1 || ok2) {
    // Do the thing
}
```

But now I have a pre-condition to be checked before the user accepting the dialog, so I added:
```
bool canOk = ComplexCondition();
BeginDisabled(!canOk);
bool ok1 = Button("OK");
bool ok2 = Shortcut(ImGuiKey_Enter, ImGuiInputFlags_None);
EndDisabled();
if (ok1 || ok2) {
    // Do the thing
}
```
The button is properly disabled, but if the user presses the "Enter" key the function is activated anyway.

In this particular case I can easily fix it by writing:
```
if (canOk && (ok1 || ok2)) {
    // Do the thing
}
```
but I wonder if `Shortcut()` should ignore the key press when in a disabled context, as I assumed it would.

If not, maybe a note in the comments would be nice, for people like me that assume too much.

**Comment by ocornut** [comment]
You are correct I think it should be disabled. I will fix it.

**Comment by ocornut** [comment]
Pushed fix fbb903e
Pushed regression test (note the 3 cases which are subtly different): https://github.com/ocornut/imgui_test_engine/commit/fee5196264106e45144c602e06ab78beef662e57
